File: src/main/java/com/synopsys/integration/detect/workflow/nameversion/DetectorProjectInfo.java
package com.synopsys.integration.detect.workflow.nameversion;

import org.jetbrains.annotations.NotNull;

import com.synopsys.integration.detector.base.DetectorType;
import com.synopsys.integration.util.NameVersion;

public class DetectorProjectInfo {
    @NotNull
    private final DetectorType detectorType;
    private final int depth;
    @NotNull
    private final NameVersion nameVersion;

    public DetectorProjectInfo(@NotNull DetectorType detectorType, int depth, @NotNull NameVersion nameVersion) {
        this.detectorType = detectorType;
        this.depth = depth;
        this.nameVersion = nameVersion;
    }

    @NotNull
    public DetectorType getDetectorType() {
        return detectorType;
    }

    public int getDepth() {
        return depth;
    }

    @NotNull
    public NameVersion getNameVersion() {
        return nameVersion;
    }
}

File: src/main/kotlin/com/synopsys/integration/detect/kotlin/nameversion/DetectorProjectInfo.kt
package com.synopsys.integration.detect.kotlin.nameversion

import com.synopsys.integration.detector.base.DetectorType
import com.synopsys.integration.util.NameVersion

class DetectorProjectInfo(val detectorType: DetectorType, val depth: Int, val nameVersion: NameVersion) {
    constructor(metadata: DetectorProjectInfoMetadata, nameVersion: NameVersion) : this(metadata.detectorType, metadata.depth, nameVersion)
}


